### PR TITLE
Feature/cloudfront access log adapter

### PR DESCRIPTION
**Pull request by fblundun**

### CODE REVIEW

**Comment by alexanderdean** [comment]
This is looking good! Something I remembered - CloudFront access log files start with two header rows:

```
#Version: 1.0
#Fields: date time x-edge-location sc-bytes c-ip cs-method cs(Host) cs-uri-stem sc-status cs(Referer) cs(User-Agent) cs-uri-query cs(Cookie) x-edge-result-type x-edge-request-id x-host-header cs-protocol cs-bytes time-taken 
```

It would be good if these were swallowed (=> None) rather than turned into bad rows...

**Comment by fblundun** [comment]
Good point! I'll implement this in the loader rather than the adapter (since adapters have to return NonEmptyLists of raw events)

**Comment by alexanderdean** [comment]
Thanks Fred...

**Comment by alexanderdean** [comment]
Also note that the exact contents of line 2 will be somewhat different across the different access log formats.

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/CloudfrontAccessLogAdapter.scala [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+ * Transforms a Cloudfront access log into raw events
+ */
+object CloudfrontAccessLogAdapter {
+
+  /**
+   * Adapter for Cloudfront web distribution access log files
+   */
+  object WebDistribution extends Adapter {
+
+    private val fieldNames = List(
```
Comment (alexanderdean) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/CloudfrontAccessLogAdapter.scala]: Capital f for FieldNames please!

**Comment by alexanderdean** [comment]
Looks great Fred! Beyond the comment about an upper-case f, can you do two things:

* Create a ticket for the Hadoop test you added
* Remove any CHANGELOG changes to prevent merge conflicts when I pull into the release branch

Once those 3 things are done, I will merge into pygmy-parrot...

**Comment by fblundun** [comment]
I moved the commits onto the latest version of release/r61-pygmy-parrot and now the CHANGELOG is unchanged.

**Comment by alexanderdean** [comment]
Nice - thanks Fred!

**Pull request merged**